File: RedSheet.DbEntities/Models/NanoRelationshipRequire.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NanoRelationshipRequires", Schema = "dbo")]
	[Serializable]
	public partial class NanoRelationshipRequire  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoRelationshipRequireId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string NanoRelationshipRequireName { get; set; } 
	[InverseProperty("NanoRelationshipRequire")]
	public ICollection<NanoScopeToNegotiateObjective> NanoScopeToNegotiateObjectives { get; set; } 


	public NanoRelationshipRequire()
	{
	this.NanoScopeToNegotiateObjectives = new HashSet<NanoScopeToNegotiateObjective>();

    }

    }
}

File: redsheet-portal/src/app/database-models/nano-relationship-require.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { NanoScopeToNegotiateObjective,  } from './'
export class NanoRelationshipRequire {
    constructor(nanoRelationshipRequire?: NanoRelationshipRequire )  {
        let properties = [ "className", "nanoRelationshipRequireId", "nanoRelationshipRequireName", "nanoScopeToNegotiateObjectives",];
        for (let property of properties)
            if (nanoRelationshipRequire && nanoRelationshipRequire[property])
                this[property] = nanoRelationshipRequire[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	nanoRelationshipRequireId : number =   0 ;
 
    @required()
    @maxLength(200)
	nanoRelationshipRequireName : string =   undefined;
	@nested(NanoScopeToNegotiateObjective)
	nanoScopeToNegotiateObjectives: NanoScopeToNegotiateObjective[];



}
